### PR TITLE
Better handling for GIF even, if not supported by the graphics library

### PR DESCRIPTION
**Pull request by annando**
When using a graphics library that can't handle GIF files or can't handle the image type at all, we don't resize it.

### CODE REVIEW

File: include/Photo.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ static function supportedTypes() {
 
 	public function __construct($data, $type=null) {
 		$this->imagick = class_exists('Imagick');
-		$this->types = $this->supportedTypes();
+		$this->types = static::supportedTypes();
```
Comment (rabuzarus) [inline include/Photo.php]: I would say this you need to say `self::` instead of `static::` (`$this->types = self::supportedTypes();`)

Comment (annando) [reply include/Photo.php]: I copied the idea from here: https://github.com/friendica/friendica/blob/develop/src/App.php#L135

Comment (annando) [reply include/Photo.php]: @rabuzarus I guess @MrPetovan could explain if this was a good idea. (I hope so)

Comment (MrPetovan) [reply include/Photo.php]: `self` always mean the class where the method is declared, however it isn't always ideal when a child class calls a parent class method. `static` always means the current class which is the intuitive behavior and should be preferred to `self`.

More on PHP's Late Static Binding: http://php.net/manual/en/language.oop5.late-static-bindings.php

File: include/Photo.php [hunk 2]
```diff
@@ -26,7 +26,7 @@ class Photo {
 	/**
 	 * @brief supported mimetypes and corresponding file extensions
 	 */
-	static function supportedTypes() {
+	static public function supportedTypes() {
```
Comment (MrPetovan) [inline include/Photo.php]: The usual order of keywords would be `public static function`.

File: include/Photo.php (hunk 1 above)
Comment (rabuzarus) [reply include/Photo.php]: @MrPetovan thanks for the explanation

File: include/Photo.php (hunk 2 above)
Comment (annando) [reply include/Photo.php]: @MrPetovan Done

**Pull request merged**